File: app/src/main/java/jp/bellware/echo/filter/FirstCut.java
package jp.bellware.echo.filter;

/**
 * 最初をカットする
 */
public class FirstCut {
    private final int first;
    private int index = 0;

    public FirstCut(int first) {
        this.first = first;
    }

    public float filter(float s){
        float d = 0;
        if(index >= first)
            d = s;
        ++index;
        return d;
    }

    public void reset() {
        index = 0;
    }
}

File: app/src/main/java/jp/bellware/echo/filter/FirstCut.kt
package jp.bellware.echo.filter

/**
 * 最初をカットする
 * @param first カット期間
 */
class FirstCut(private val first: Int) {
    private var index = 0

    /**
     * フィルターをかける
     */
    fun filter(s: Float): Float {
        var d = 0f
        if (index >= first)
            d = s
        ++index
        return d
    }

    /**
     * リセット
     */
    fun reset() {
        index = 0
    }
}
